### ISSUE TITLE
Help - Looking for ideas to implement a vault

### ISSUE THREAD

**Issue by arvind47d** [yunabe/tslab#22]
Hi, Can you please let me know if you have ideas to implement a secure store? 

1. Setup a vault and add secrets
2. From Jupyter UI, tell the kernel to pick the secret possibly replace in a placeholder
    for example  if cell has 
`const header = { auth: bearer $${get_secret("xyz")} }`
3. Kernel preprocesses code extracts $${get_secret("xyz")}, securely retries it from the vault and replaces the secret in the code and continue execution of the code

Thanks for your insights! Great kernel by the way.

**Comment by yunabe** [comment]
I'm not familiar with Vault (Do you mean https://www.vaultproject.io/, right?). But I guess when you set up "Vault" in step(1), Vault stores a credential somewhere and we may be able to read it by Node.js File API. If accessing Vault from Node.js is common, I think there is an existing package of Vault client.

For (2) and (3), I'm not sure why we need `$${}` in your proposal. If `get_secret("xyz")` is JavaScript, I feel it just works w/o `$${...}` as JavaScript.
Also, the new [*Template String* syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) might be helpful in your use case.

**Comment by arvind47d** [comment]
yes, that's the vault. The thought here is to not to allow exposing the secret values through code to the UI (via a get)

**Comment by yunabe** [comment]
I still don't understand what is the key of your question.

What is the problem of pseud-code below?

```javascript
const header = { auth: bearer, xyz: vault.read("xyz") }
```

I don't think the current `tslab` implementation does not print the content of `header` automatically.
